modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/BindingsRegistry.java

```diff
@@ -43,38 +51,58 @@
 
     private final BindingDeleteListener bindingDeleteListener;
 
+    private List<BindingsRegistryListener> bindingsRegistryListeners;
+
+    private final ReadWriteLock lock;
```

Comment:
Why have we introduced a lock in this class?

Reply:
Fixed in cd6b801